tests/unittests/test_handler/test_handler_reset_rmc.py

```diff
@@ -0,0 +1,54 @@
+from cloudinit.config import cc_reset_rmc
+
+from cloudinit import util
+
+from cloudinit.config.cc_write_files import write_files
+from cloudinit.tests import helpers as t_help
+from cloudinit.tests.helpers import mock
+
+import logging
+import os
+
+LOG = logging.getLogger(__name__)
+NODE_FILE = '/etc/ct_node_id'
+RSCT_PATH = '/opt/rsct/install/bin'
+
+
+class TestRsctNodeFile(t_help.FilesystemMockingTestCase):
+    def setUp(self):
+        super(TestRsctNodeFile, self).setUp()
+        self.tmp = self.tmp_dir()
+
+    def recfgct(self):
+        content = '34fhs7kl892840f45'
+        write_files(
+            "test_rsct_node", [{"content": content, "path": NODE_FILE}])
+        return util.load_file(NODE_FILE)
+
+    def rmcctrl(self):
+        return ('', '')
+
+    @mock.patch(
+        'cloudinit.config.cc_reset_rmc.reconfigure_rsct_subsystems',
+        side_effect=recfgct)
+    @mock.patch(
+        'cloudinit.config.cc_reset_rmc.rmcctrl',
+        side_effect=rmcctrl)
+    def test_rsct_node(self, reconfigure_rsct_subsystems, rmcctrl):
+        self.allowed_subp = True
+        reconfigure_rsct_subsystems.return_value = \
+            cc_reset_rmc.reconfigure_rsct_subsystems()
+        rmcctrl.return_value = cc_reset_rmc.rmcctrl()
+        content = '234fddkshfi43248'
+        write_files(
+            "test_rsct_node", [{"content": content, "path": NODE_FILE}])
+        node_id_before = util.load_file(NODE_FILE)
+        cc_reset_rmc.reset_rmc()
+        node_id_after = util.load_file(NODE_FILE)
+        self.assertNotEqual(node_id_before, node_id_after)
+
+    def test_path(self):
+        self.allowed_subp = True
+        orig_path = os.environ.get('PATH')
+        x = cc_reset_rmc.add_path(orig_path)
```

Comment:
you can actually mock the os.environ dictionary. that way your test case does not permenantly change the environment (as it does right now).

you can see that in cloudinit/tests/test_dhclient_hook.py or cloudinit/tests/test_subp.py.